Makefile

```diff
@@ -108,11 +110,31 @@ build-containers: migration-planner-api-container migration-planner-agent-contai
 
 .PHONY: build-containers
 
-push-containers: build-containers
-	$(PODMAN) push $(MIGRATION_PLANNER_API_IMAGE):latest
-	$(PODMAN) push $(MIGRATION_PLANNER_API_IMAGE):$(REGISTRY_TAG)
-	$(PODMAN) push $(MIGRATION_PLANNER_AGENT_IMAGE):latest
-	$(PODMAN) push $(MIGRATION_PLANNER_AGENT_IMAGE):$(REGISTRY_TAG)
+quay-login:
+	if [ ! -f "$(DOCKER_AUTH_FILE)" ] && [ -n "$(QUAY_USER)" ] && [ -n "$(QUAY_TOKEN)" ]; then \
+		mkdir -p "$(DOCKER_CONF)"; \
+		$(PODMAN) login --authfile "$(DOCKER_AUTH_FILE)" -u="$(QUAY_USER)" -p="$(QUAY_TOKEN)" quay.io; \
+	fi;
+
+push-api-container: migration-planner-api-container quay-login
+	if [ -f "$(DOCKER_AUTH_FILE)" ]; then \
+		$(PODMAN) push $(MIGRATION_PLANNER_API_IMAGE):latest --authfile $(DOCKER_AUTH_FILE); \
```

Comment:
If we have run `podman login` do we really need to pass `authfile` again? I think the creds are stored in cache

Reply:
Yes, because we hold the token localy and not in the default file

Reply:
can u maybe simplify it so it's like:

```
DOCKER_AUTH_FLAG="--autfile ${DOCKER_AUTH_FILE}"
$(PODMAN) push $(MIGRATION_PLANNER_API_IMAGE):latest $(DOCKER_AUTH_FLAG)
```

or anything similar so we don't need if else?

Reply:
I have actually tried to do this optimization, but from some reason the `DOCKER_AUTH_FLAG` was always empty (initiating it with empty string and than setting the flag if the file exists)